src/Context/BaseUrlContext.tsx

```diff
@@ -0,0 +1,18 @@
+import { createContext, ReactNode } from "react";
+
+export const BaseUrlContext = createContext("");
+
+// ReactNode => hook for react used with children
+export default function BaseUrlContextProvider({
+	children,
+}: {
+	children: ReactNode;
+}) {
+	const baseUrl = "https://upskilling-egypt.com:3003/api/v1";
```

Comment:
Consider looking for the alternative approaches that I mentioned.
- Creating AxiosInstance
or
- urls.ts file